src/main/java/gregtech/client/renderer/pipe/LaserPipeRenderer.java

```diff
@@ -0,0 +1,89 @@
+package gregtech.client.renderer.pipe;
+
+import codechicken.lib.render.CCRenderState;
+import codechicken.lib.render.pipeline.IVertexOperation;
+import codechicken.lib.vec.Cuboid6;
+import codechicken.lib.vec.uv.IconTransformation;
+import gregtech.api.pipenet.block.BlockPipe;
+import gregtech.api.pipenet.block.IPipeType;
+import gregtech.api.pipenet.tile.IPipeTile;
+import gregtech.api.unification.material.Material;
+import gregtech.api.util.GTUtility;
+import gregtech.client.renderer.texture.Textures;
+import gregtech.client.utils.BloomEffectUtil;
+import gregtech.common.ConfigHolder;
+import gregtech.common.pipelike.laser.LaserPipeType;
+import gregtech.common.pipelike.laser.tile.TileEntityLaserPipe;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.texture.TextureMap;
+import net.minecraft.util.BlockRenderLayer;
+import net.minecraft.util.EnumFacing;
+import org.apache.commons.lang3.ArrayUtils;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.EnumMap;
+
+public class LaserPipeRenderer extends PipeRenderer {
+    public static final LaserPipeRenderer INSTANCE = new LaserPipeRenderer();
+    private final EnumMap<LaserPipeType, TextureAtlasSprite> pipeTextures = new EnumMap<>(LaserPipeType.class);
+    private boolean active = false;
+    public LaserPipeRenderer() {
+        super("gt_laser_pipe", GTUtility.gregtechId("laser_pipe"));
+    }
+
+    @Override
+    public void registerIcons(TextureMap map) {
+        pipeTextures.put(LaserPipeType.NORMAL, Textures.LASER_PIPE_IN);
+    }
+
+    @Override
+    public void buildRenderer(PipeRenderContext renderContext, BlockPipe<?, ?, ?> blockPipe, @Nullable IPipeTile<?, ?> pipeTile, IPipeType<?> pipeType, @Nullable Material material) {
+        if (pipeType instanceof LaserPipeType) {
+            renderContext.addOpenFaceRender(new IconTransformation(pipeTextures.get(pipeType)))
+                    .addSideRender(false, new IconTransformation(Textures.LASER_PIPE_SIDE))
+                    .addSideRender(new IconTransformation(Textures.LASER_PIPE_OVERLAY));
+            active = !ConfigHolder.client.preventAnimatedCables && pipeTile instanceof TileEntityLaserPipe laserPipe && laserPipe.isActive();
+        }
+    }
+
+    @Override
+    protected void renderOtherLayers(BlockRenderLayer layer, CCRenderState renderState, PipeRenderContext renderContext) {
```

Comment:
Was there any way to potentially re-use some of the existing pipe rendering methods here, such as `renderPipeCube` or `renderPipeSide`?